test_runner.cmd

```diff
@@ -0,0 +1,2 @@
+@ECHO OFF
+@ruby.exe -x "%~dpn0" %*
```

Comment:
What magic is this? 😆

Reply:
Windows fun!

Actually, something very similar is generated by RubyGems when it creates Windows bin stubs.

With *nix systems, one has executable permissions and shebang lines.

On Windows, the extension determines whether a file is executable.

`%*` is equivalent to `ARGV`, and `%~dpn0` is equivalent to `File.basename(__FILE__, '.*')`

So, in this instance it is equivalent to `ruby.exe -x "test_runner" ARGV`